src/BridgeFinder.ts

```diff
@@ -169,14 +169,24 @@ export class BridgeFinder extends (EventEmitter as new () => TypedEmitter<Bridge
         }
 
         let bridgeID;
-        const hostname = await this.getHostnameFromIP(ipaddr);
-        logDebug('got hostname from IP:', hostname);
+        let hostname;
+
         try {
+            hostname = await this.getHostnameFromIP(ipaddr);
+            logDebug('got hostname from IP:', hostname);
+
             // If the format of the hub hostname changes, this match can break,
             // and will appear as your credentials not working any longer
             bridgeID = hostname!.match(/[Ll]utron-(?<id>\w+)\.local/)!.groups!.id;
         } catch {
-            bridgeID = ipaddr.replace('.', '_');
+            logDebug("got exception looking up hostname");
+            if (hostname) {
+                bridgeID = ipaddr.replace('.', '_');
+            } else if (this.secrets.size == 1) {
+                // If there is only one hub then we can get the bridge value from the
+                // secrets
+                bridgeID = this.secrets.entries().next().value[0];
```

Comment:
This would be better if it were explicit instead of implicit. Refactor to check for ambiguity and emit an error if that's the case.

Reply:
No problem, I'll refactor these methods. Now that I think about it more what do you think about defaulting to the single bridge case and only doing the DNS query if there is more than one bridge? This seems logical and will simply the error handling. Thoughts?

Reply:
Yeah that is actually the most straightforward approach, I like it!

Reply:
I think you can even skip starting the bridge finder altogether if there’s only one secret

Reply:
From my understanding of this it seems like the bridge finder is still required since the leap client initializes with an IP address. My plan is to update the PR with some possible changes.

Reply:
Oh! Derp. Right, sorry.

Reply:
OK I updated the PR, let me know what you think.